the-toad-tribune/src/SubApp.tsx

```diff
@@ -60,16 +64,36 @@ const SubApp = () => {
     );
 
     const moviesEverythingRequest = new NewsEverythingRequest({ q: "movies" });
-    getNewsEverything(moviesEverythingRequest).then((res) => setMoviesArticle(res));
-
-    const stonksHeadlinesRequest = new NewsEverythingRequest({q: "stocks"});
-    getNewsEverything(stonksHeadlinesRequest).then((res) => 
-      setStonksArticle(res));
+    getNewsEverything(moviesEverythingRequest).then((res) =>
+      setMoviesArticle(res)
+    );
 
+    const stonksHeadlinesRequest = new NewsEverythingRequest({ q: "stocks" });
+    getNewsEverything(stonksHeadlinesRequest).then((res) =>
+      setStonksArticle(res)
+    );
   };
 
   const adsDataGrabber = () => {
-    getAds().then(res => setAds(res));
+    getAds().then((res) => setAds(res));
+  };
+
+  const renderMoreInfoPage = () => {
+    setIsMainLayout(false);
+    setIsSearchPage(false);
+    setIsMoreInfo(true);
+  };
+
+  const renderSearchPage = () => {
+    setIsMainLayout(false);
+    setIsSearchPage(true);
+    setIsMoreInfo(false);
+  };
+
+  const renderMainLayoutPage = () => {
+    setIsMainLayout(true);
+    setIsSearchPage(false);
+    setIsMoreInfo(false);
```

Comment:
Possible DRY solution?
```
const renderPage = (param1, param2, param3) => {
     setLayout1(param1)
     setLayout2(param2)
     setLayout3(param3)
}
```

Reply:
I like that! Thanks for the refactor idea!